### PR TITLE
Database integration, test environment fixes and documentation

### PR DESCRIPTION
**Pull request by eedrummer**
Updated the main README and pushed some detail into subproject READMEs. Still need to create READMEs for all subprojects.

Integrated the `order-review` hook with the database. Introduced a utility method to calculate a patient's age.

I have disables fetching in `server` and login in `request-generator` for now as I have been unable to get them to work with my Keycloak setup.

Several changes to the test environment:
* Now using an in-memory database
* Gradle is set up to find all `@Test` annotations. Previous setup would only find `org.junit.jupiter.api.Test` annotated methods and not `org.junit.Test` and never tell you... which is awesome.

`request-generator` contains some fixes:
* URLs changed to match new endpoints
* Request now uses values from the form

### CODE REVIEW

File: server/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+This will start the server running on http://localhost:8090.
+
+## Server endpoints
+|Relative URL|Endpoint Description|
+|----|----|
+|`/`|Web page with basic RI information|
+|`/data`|Web-based administrative interface|
+|`/cds-services/order-review-crd`|CDS Hook endpoint for order-review|
+|`/cds-services/medication-prescribe-crd`|CDS Hook endpoint for medication-prescribe|
+
```
[the remark this answers is not in the document]
Comment (eedrummer) [reply server/README.md]: Fixed

File: server/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+`gradle bootRun`
+
+This will start the server running on http://localhost:8090.
+
+## Server endpoints
+|Relative URL|Endpoint Description|
+|----|----|
+|`/`|Web page with basic RI information|
+|`/data`|Web-based administrative interface|
```
[the remark this answers is not in the document]
Comment (eedrummer) [reply server/README.md]: Yes, but I don't think that changes the endpoint. It looks like we have the admin interface set up now to require OAuth, right?

[the remark this answers is not in the document]
Comment (KeeyanGhoreshi) [reply server/README.md]: Pretty much anything can be made to require OAuth by changing `SecurityConfig.java` in the server project.

[the remark this answers is not in the document]
Comment (daco101) [reply server/README.md]: Ok, we could consider changing SecurityConfig to have something like `antMatchers("/data", "/data/*")` at some point.

**Pull request merged**